src/gasman.h

```diff
@@ -293,13 +298,59 @@ EXPORT_INLINE UInt SIZE_BAG_CONTENTS(const void *ptr) {
 **  the application  must inform {\Gasman}  that it  has changed  the bag, by
 **  calling 'CHANGED_BAG(old)' in the above example (see "CHANGED_BAG").
 */
+#ifdef HPCGAP
+EXPORT_INLINE PURE_FUNC int ReadCheck(Bag bag)
+{
+    Region *region;
+    region = REGION(bag);
+    if (!region)
+        return 1;
+    if (region->owner == GetTLS())
+        return 1;
+    if (region->readers[TLS(threadID)])
+        return 1;
+    return 0;
+}
+
+EXPORT_INLINE PURE_FUNC int WriteCheck(Bag bag)
+{
+    Region *region;
+    region = REGION(bag);
+    return !region || region->owner == GetTLS();
+}
+
+extern volatile int GuardDummy;
+extern PURE_FUNC int HandleReadGuardError(Bag);
+extern PURE_FUNC int HandleWriteGuardError(Bag);
+#endif // HPCGAP
+
 EXPORT_INLINE Bag *PTR_BAG(Bag bag)
+{
+    GAP_ASSERT(bag != 0);
+#ifdef HPCGAP
+    if (!WriteCheck(bag))
+      GuardDummy = HandleWriteGuardError(bag);
+#endif
+    return *(Bag**)bag;
```

Comment:
```suggestion
    return *(Bag **)bag;
```

`git clang-format master` ?

Reply:
I'm not sure I understand. This is the same format as it currently exists on the master branch?

Reply:
The old code is indeed not (yet) consistently clang-formatted; but you are adding new code here, which you may as well clang-format. Anyway, this is hardly important, feel free to ignore :-)